src/Common/Query/GeocodeQuery.php

```diff
@@ -46,12 +46,12 @@
     /**
      * @var array
      */
-    private $data = [];
+    protected $data = [];
 
     /**
      * @param string $text
      */
-    private function __construct(string $text)
```

Comment:
Why do you need to override a private constructor? Why not override the `create` function?

Reply:
It seemed to me to be logical that when the class is not `final` any more, child classes should be free to do a different `__construct`. But I can do as you say, and override the create function instead, no problem.

Reply:
I would like to keep the changes to the GeocodeQuery to a strict minimum.